apps/examples/binary_update/binary_update_main.c

```diff
@@ -78,20 +110,130 @@ static void binary_update_getinfo_all(void)
 	}
 }
 
-static void binary_update_getinfo(char *name)
+static void binary_update_info(binary_info_t *pre_bin_info, binary_info_t *cur_bin_info, int condition)
+{
+	printf(" === [%4s] Update info === \n", cur_bin_info->name);
+	printf(" %4s | %8s | %s \n", "Con", "Version", "Inactive part");
+	printf(" ------------------------- \n");
+	printf(" %4s | %8s | %s \n", "Pre", pre_bin_info->version, pre_bin_info->dev_path);
+	printf(" %4s | %8s | %s \n", "Cur", cur_bin_info->version, cur_bin_info->dev_path);
+	printf(" ========================= \n");
+
+	if (condition == DOWNLOAD_SUCCESS) {
```

Comment:
It's hard to understand, also you have to check success case even if download failed.
Also you should display more correct info.(you called strncmp with path but display too simply) then how about below style?
```
if (condition == DOWNLOAD_SUCCESS) {
    printf("Binary Download Success!\n");
} else {
   printf("Binary Download Failed!\n");
}

if (strncmp() ==0) {
    printf("Binary path is Same!\n");
} else {
    printf("Binary path is not Same! prev path : %s cur path : %s\n", priv, cur);
}

Reply:
Thanks. I will think more.